src/dataflow/src/sink/kafka.rs

```diff
@@ -566,17 +690,185 @@ impl KafkaSinkState {
         Err(last_error)
     }
 
+    async fn determine_latest_consistency_record(
+        &self,
+    ) -> Result<Option<Timestamp>, anyhow::Error> {
+        // Return the list of partition ids associated with a specific topic
+        /// Polls a message from a Kafka Source
+        fn get_next_message(
+            consumer: &mut BaseConsumer,
+            timeout: Duration,
+        ) -> Result<Option<(Vec<u8>, i64)>, anyhow::Error> {
+            if let Some(result) = consumer.poll(timeout) {
```

Comment:
Is this a blocking call?

Reply:
yes -- it's only ever called on a background thread.  It'd be tricky to asyncify because rdkafka doesn't expose an async interface.  Calling it with `tokio::spawn_blocking` seems like overkill / needlessly complex since it's only ever called on one of those threads anyways -- do you think a comment is sufficient?